mineinabyss-features/src/main/kotlin/com/mineinabyss/armorstandlock/ArmorStandLockingListener.kt

```diff
@@ -92,4 +99,114 @@ class ArmorStandLockingListener : Listener {
             isCancelled = true
         }
     }
+
+    @EventHandler
+    fun EntityMoveEvent.onEntityMove() {
+        /**
+         * This listener is used to prevent gravity and water from affecting armor stands.
+         * This is done by setting the entity's velocity to 0 and cancelling the event.
+         */
+        if (entityType != EntityType.ARMOR_STAND) return
+
+        val armorStand = entity.toGeary().get<LockArmorStand>() ?: return
+
+        if (!armorStand.lockState) return
+
+        // We only care about ArmorStands that have fishing hooks as passengers
+        if (entity.passengers.none { it.type == EntityType.FISHING_HOOK }) return
+
+        entity.velocity = Vector(0.0, 0.0, 0.0)
+        isCancelled = true
+    }
```

Comment:
This can probably be removed entirely in favor of simply spawning the armorstand with setGravity(false).

Reply:
On second thought that wouldn't be the best solution as they're meant to spawn as normal armorstands.
So on toggling it set the gravity = to togglestate inside ArmorStandLockingFeature

Reply:
This is also used to prevent Water from flowing Armorstands away. As far as I know, giving them no gravity won't affect water flow

Reply:
water only affects armorstands if theyre dropped onto a flowing stream.
So if it doesnt have gravity it should not be able to flow

From testing it rn it also seems to not move when water flows onto it after gravity is disabled
![image](https://user-images.githubusercontent.com/62521371/163680266-0909c3d5-888b-476b-ac25-a311383ac3af.png)

Reply:
That's suprising. Thanks!